src/main/java/de/tum/in/www1/artemis/domain/ProgrammingExerciseBuildConfig.java

```diff
@@ -0,0 +1,277 @@
+        this.allowBranching = allowBranching;
+    }
+
+    /**
+     * We store the build plan configuration as a JSON string in the database, as it is easier to handle than a complex object structure.
+     * This method parses the JSON string and returns a {@link Windfile} object.
+     *
+     * @return the {@link Windfile} object or null if the JSON string could not be parsed
+     */
+    public Windfile getWindfile() {
+        if (buildPlanConfiguration == null) {
+            return null;
+        }
+        try {
+            return Windfile.deserialize(buildPlanConfiguration);
+        }
+        catch (JsonProcessingException e) {
+            log.error("Could not parse build plan configuration for programming exercise {}", this.getId(), e);
+        }
+        return null;
+    }
+
+    public void filterSensitiveInformation() {
+        setBuildPlanConfiguration(null);
+        setBuildScript(null);
+    }
+
+    public boolean isTestwiseCoverageEnabled() {
+        return testwiseCoverageEnabled;
+    }
+
+    public void setTestwiseCoverageEnabled(boolean testwiseCoverageEnabled) {
+        this.testwiseCoverageEnabled = testwiseCoverageEnabled;
+    }
+
+    public ProgrammingExercise getProgrammingExercise() {
+        return programmingExercise;
+    }
+
+    public void setProgrammingExercise(ProgrammingExercise programmingExercise) {
+        this.programmingExercise = programmingExercise;
+    }
+
+    public boolean hasBuildPlanAccessSecretSet() {
+        return buildPlanAccessSecret != null && !buildPlanAccessSecret.isEmpty();
+    }
+
+    @Nullable
+    public String getBuildPlanAccessSecret() {
+        return buildPlanAccessSecret;
+    }
+
+    public void generateAndSetBuildPlanAccessSecret() {
+        buildPlanAccessSecret = UUID.randomUUID().toString();
+    }
+
+    /**
+     * Copy the build config from the originalBuildConfig to this build config
+     *
+     * @param originalBuildConfig the original build config
+     */
+    public void copyBuildConfig(ProgrammingExerciseBuildConfig originalBuildConfig) {
+        this.setBranch(originalBuildConfig.getBranch());
+        this.setBuildPlanConfiguration(originalBuildConfig.getBuildPlanConfiguration());
+        this.setCheckoutPath(originalBuildConfig.getCheckoutPath());
+        this.setCheckoutSolutionRepository(originalBuildConfig.getCheckoutSolutionRepository());
+        this.setDockerFlags(originalBuildConfig.getDockerFlags());
+        this.setSequentialTestRuns(originalBuildConfig.hasSequentialTestRuns());
+        this.setBuildScript(originalBuildConfig.getBuildScript());
+        this.setTestwiseCoverageEnabled(originalBuildConfig.isTestwiseCoverageEnabled());
+        this.setTimeoutSeconds(originalBuildConfig.getTimeoutSeconds());
+        this.setTheiaImage(originalBuildConfig.getTheiaImage());
+        this.setAllowBranching(originalBuildConfig.isAllowBranching());
+        this.setBranchRegex(originalBuildConfig.getBranchRegex());
+    }
```

Comment:
Can you instead use a Copy-Constructor? Since this is a void you currently first need to create a new object before then using this method.